msl/equipment/resources/vaisala/ptu300.py

```diff
@@ -0,0 +1,199 @@
+            'hPa', 'psia', 'inHg', 'torr', 'bara', 'barg', 'psig', 'mbar', 'mmHg', 'kPa', 'Pa', 'mmH2O', 'inH2O'
+        ]
+        check_string_m = "Output units   : metric"
+        available_units = self.query("UNIT")
+        if not available_units == check_string_m:  # confirming device is of type PTU300
+            self.log_error("Check correct device connected")
+            return
+
+        for quantity, u in desired_units.items():
+            if quantity == "RH":    # only option is %RH
+                self._units["RH"] = "%RH"
+
+            elif "T" in quantity:   # options are °C, °F
+                if 'F' in u:        # Temperature and humidity setting is done via metric or 'non metric'
+                    celcius = False
+                if celcius:
+                    r_m = self.query("UNIT M")
+                    self._units["T"] = "ºC"
+                else:
+                    r_m = self.query("UNIT N")
+                    self._units["T"] = "ºF"
+
+                if not (r_m == check_string_m) == celcius:
+                    self._units["T"] = None
+                    self.check_for_errors()
+
+            elif u in allowed_units:  # assume this is a pressure quantity based on the unit
+                if not quantity == "P" and "P" in desired_units:
+                    self.log_info(f"Using pressure setting for P for quantity {quantity}")
+                else:
+                    r_p = self.query(f"UNIT P {u}")
+                    check_string_p = f"P units        : {u}"
+                    if not r_p == check_string_p:
+                        self.check_for_errors()
+                    self._units[quantity] = u
+            else:  # quantity is not pressure, temperature or humidity, so ask user to set the unit manually
+                self.log_error(f"{u} is not able to be set for {quantity}. Please set this unit manually.")
+
+    @property
+    def units(self) -> dict[str, str]:
+        """A dictionary of measured quantities and their associated units as set on the device."""
+        return self._units
+
+    def set_format(self, format: str) -> bool:
+        """Set the format of data output to follow the pattern in the format string.  For example, in the format string
+        '4.3 P " " 3.3 T " " 3.3 RH " " SN " " #r #n', x.y is the number of digits and decimal places of the values;
+        P, T, RH, and SN are placeholders for pressure, temperature, relative humidity, and serial number values;
+        and " ", #r, and #n represent a string constant, carriage-return, and line feed respectively.
+        Additional allowed modifiers include ERR for error flags, U5 for unit field and (optional) length,
+        TIME for time as [hh:mm:ss], and DATE for date as [yyyy-mm-dd]. For more options, refer to the manual.
+
+        :param format: string representing desired output format
+        :return: bool to indicate successful setting of format
+        """
```

Comment:
Probably better to use a literal string in the docstring
```rst
For example, in the format string ``'4.3 P " " 3.3 T " " 3.3 RH " " SN " " #r #n'``
```
Try building the docs to see how it renders.